cli/azd/cmd/auth_login.go

```diff
@@ -254,81 +257,103 @@ func (la *loginAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+		// If this does fail, we want the user to be aware of this. Like other stored azd account data,
+		// stored subscriptions are currently tied to the OS user, and not the individual account being logged in,
+		// this means cross-contamination is possible.
+		if err := la.accountSubManager.ClearSubscriptions(ctx); err != nil {
+			return nil, err
+		}
+
+		err := la.accountSubManager.RefreshSubscriptions(ctx)
 		if err != nil {
-			log.Printf("failed to check auth status: %s", err.Error())
-			res.Status = contracts.LoginStatusUnauthenticated
-		} else {
-			res.Status = contracts.LoginStatusSuccess
-			res.ExpiresOn = &token.ExpiresOn
+			// If this fails, the subscriptions will still be loaded on-demand.
+			// erroring out when the user interacts with subscriptions is much more user-friendly.
+			log.Printf("failed retrieving subscriptions: %v", err)
+		}
+	} else {
+		// Service principals do not typically require subscription caching (running in CI scenarios)
+		// We simply clear the cache, which is much faster than rehydrating.
+		err := la.accountSubManager.ClearSubscriptions(ctx)
+		if err != nil {
+			log.Printf("failed clearing subscriptions: %v", err)
 		}
 	}
 
-	if !la.flags.onlyCheckStatus && res.Status == contracts.LoginStatusSuccess {
-		// Rehydrate or clear the account's subscriptions cache.
-		// The caching is done here to increase responsiveness of listing subscriptions (during azd init).
-		// It also allows an implicit command for the user to refresh cached subscriptions.
-		if la.flags.clientID == "" {
-			// Deleting subscriptions on file is very unlikely to fail, unless there are serious filesystem issues.
-			// If this does fail, we want the user to be aware of this. Like other stored azd account data,
-			// stored subscriptions are currently tied to the OS user, and not the individual account being logged in,
-			// this means cross-contamination is possible.
-			if err := la.accountSubManager.ClearSubscriptions(ctx); err != nil {
-				return nil, err
-			}
+	la.console.Message(ctx, cLoginSuccessMessage)
+	return nil, nil
+}
 
-			err := la.accountSubManager.RefreshSubscriptions(ctx)
-			if err != nil {
-				// If this fails, the subscriptions will still be loaded on-demand.
-				// erroring out when the user interacts with subscriptions is much more user-friendly.
-				log.Printf("failed retrieving subscriptions: %v", err)
-			}
-		} else {
-			// Service principals do not typically require subscription caching (running in CI scenarios)
-			// We simply clear the cache, which is much faster than rehydrating.
-			err := la.accountSubManager.ClearSubscriptions(ctx)
-			if err != nil {
-				log.Printf("failed clearing subscriptions: %v", err)
-			}
-		}
+// Verifies that the user has credentials stored,
+// and that the credentials stored is accepted by the identity server (can be exchanged for access token).
+func (la *loginAction) verifyLoggedIn(ctx context.Context) (*azcore.AccessToken, error) {
```

Comment:
Should we have new unit tests for this method?

Reply:
Do you mean integration/component testing? The function itself is fairly straightforward refactored from existing.

I'll think about how we can test login better though.

Reply:
I've added explicit tests to `login` using service principal. This was already implicitly tested during the pipeline run, but the changes in this direction is good.

I've also started separating out env variables from the test configuration. The V1 is a minimal change. We may want to refactor Live test config separately in the future, but it was the simplest change to write the test without feeling directly coupled.